index.js

```diff
@@ -97,6 +103,39 @@ io.sockets.on('connection', function (socket){
 
   });
 
+  socket.on('generateToken', function (data, callback) {
+    if (socket.username === undefined || socket.username === null) {
+      return callback('userNotJoined');
+    }
+    // First generate the JWTID
+    crypto.randomBytes(48, function(err, buffer) {
+      if (err) {
+        return callback(err);
+      }
+      var tokenid = buffer.toString('hex');
+      var token = null;
+      try {
+        // Try to sign teh token
+        token = jwt.sign(
+          {
+            userID: socket.username,
+            appID: config.appID
+          }, privKey,
+          {
+            algorithm: "ES256",
+            jwtid: tokenid,
+            expiresIn: 300, //5 minutes
+            notBefore: -300 //-5 minutes
+          });
+      } catch (error) {
+        console.log(error);
+        return callback(error);
+      }
+      console.log({action: "GrantToken", user: socket.username, tokenid: tokenid});
```

Comment:
In real system, for audit purposes you should log the generated tokens. As this is a sample implementation, we will just log the fact that the token was granted.